packages/pigeon/lib/swift_generator.dart

```diff
@@ -185,27 +185,27 @@ import FlutterMacOS
           if (!hostDatatype.isBuiltin &&
               customClassNames.contains(field.type.baseName)) {
             indent.writeln('var ${field.name}: $fieldType? = nil');
-            indent.write('if let ${field.name}List = $listValue as? [Any?] ');
+            indent.write('if let ${field.name}List = $listValue as? [Any] ');
             indent.addScoped('{', '}', () {
               indent.writeln(
-                  '${field.name} = $fieldType.fromList(${field.name}List)');
+                  '${field.name} = $fieldType.fromList(${field.name}List as [Any])');
             });
           } else if (!hostDatatype.isBuiltin &&
               customEnumNames.contains(field.type.baseName)) {
             indent.writeln('var ${field.name}: $fieldType? = nil');
-            indent.write('if let ${field.name}RawValue = $listValue as? Int ');
+            indent.write('if let ${field.name}RawValue = $listValue as! Int? ');
             indent.addScoped('{', '}', () {
               indent.writeln(
                   '${field.name} = $fieldType(rawValue: ${field.name}RawValue)');
             });
```

Comment:
Since we're force-casting, can't we get rid of the whole `if let` construction? I.e., replace all of this with:
```
var $field.name: $fieldType? = $listValue as! Int?
```

and similarly in for the other types?